libmscore/chordlist.cpp

```diff
@@ -1464,6 +1464,122 @@ ChordDescription::ChordDescription(const QString& name)
       exportOk = false;
       }
 
+//---------------------------------------------------------
+//   intervals
+//    returns a QList of integers holding the intervals
+//    from the root note
+//---------------------------------------------------------
+QList<int> ChordDescription::intervals() const {
+      //TODO - PHV take a look again later, this might be uneeded
+      QList<int> ret;
+      int k = chord.getKeys();
+      int n = 1;
+      k >>= 1;
+      while (k > 0) {
+            if (k & 1)
+                  ret << n;
+            n++;
+            k >>= 1;
+            }
+      return ret;
+      }
+
+//---------------------------------------------------------
+//   noteTpcs
+//    returns a QList of note tpcs in the chord ordered by
+//    pitch
+//---------------------------------------------------------
+QList<int> ChordDescription::noteTpcs(int rootTpc) const {
+      //TODO - PHV take a look again later, this isn't super efficient
+      QList<int> ret;
+
+      static const HChord dimChord(0,3,6,9);
+
+      //make sure diminished chord has a diminished 7th
+      if (chord == dimChord) {
+            ret << tpcInterval(rootTpc, 3, -1) << tpcInterval(rootTpc, 5, -1)
+                << tpcInterval(rootTpc, 7, -2);
+            return ret;
+            }
+
+      if (chord.contains(3)) {
+            if (!chord.contains(4))
+                  //minor 3rd
```

Comment:
here and in various other places, that comment should better be appended to either the line above or below, fo make clear that it is a one-line if statement and not starting a block

Reply:
Oh that makes sense thank you